### DISCUSSION TITLE
VolumeReconstruction module: how can the output volume be saved?

### DISCUSSION THREAD

**Discussion by SenonETS** [PlusToolkit/PlusLib#821] [Q&A]
Hello PLUS team,

I am following this [tutorial video](https://www.youtube.com/watch?v=2vXeJxYFou4), trying to generate Ultrasound volume.

However, in the Volume Reconstruction module, I was not able to find any function to save the volume, or rendering the three anatomy planes of the volume. 

Here is [a screenshot](https://ibb.co/kqYb7MK) of what I have done.

As shown in the screenshot, the output "volume" is a created new volume. I have tried to save/rendering it after "start" reconstruction, after "stop" reconstruction; however, the input "Image_Transducer" image can never be collected by the "Volume."

I can currently only save an empty volume. 

Could anyone please share some idea about how to collect 2D scans into the volume, and then save it as an ultrasound volume data (.nrrd) file?

**Comment by Sunderlandkyl** [comment]
Are there any error messages in the log? Also can you try doing volume reconstruction with an untransformed ROI?

**Reply by SenonETS** [reply]
Hi Sunderlankyl,

It is so great to see your reply!

I do not know how or where to check the log message of Slicer.
And I was not able to rotate the ROI. It seems that there is no option of rotating the ROI, which is not convenient. Could you please let me know how if there is an option to rotate the ROI? 

The ROI always follows the bounding box, while there is neither way to rotate the bounding box. The 2D scan always sits on the **diagonal** surface of the bounding box, which results in a super large ROI to **diagonally** cover the real region of interest and thus is not practical (with most space wasted) for an ultrasound transducer; that is why I tried the transform, even though the transform does not allow me to rotate using the mouse neither. 

Also, I was not able to find undo / redo function in Slicer, which is not convenient when I did something wrong.
Could you please let me know how to undo a step in Slicer?

I did try doing volume reconstruction with an untransformed ROI, and here is [the screenshot](https://ibb.co/0tvRw0t).
Still, I was not able to get any non-empty volume.

Please help me.

**Reply by Sunderlandkyl** [reply]
You can see the error log in the application menu bar: View->Error log, or by clicking on the icon in the bottom right corner.

There is currently no undo/redo feature in Slicer except in specific modules where it is implemented (Segment editor).

**Reply by SenonETS** [reply]
Hi Sunderlankyl,

I tried again, with untransformed ROI, and checked the Error log, as shown in [this screenshot](https://ibb.co/mbfRNzF).

It seems that everything is working very well, with neither error nor warning, except the output volume is always empty.

Could you please help have a check?
